### PR TITLE
Use hooks instead of containers with render props in product mutations

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it simplifies code and removes obsolete pattern. Also makes life easier when you want to type return functions as `() => Promise` instead of `() => void`.

**PR intended to be tested with API branch:** master

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [ ] Number of API calls is optimized.
1. [ ] The changes are tested.
1. [ ] Data-test are added for new elements.
1. [x] Type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Review by orzechdev — APPROVED** [review]

File: src/products/views/ProductCreate.tsx [hunk 1]
```diff
@@ -53,118 +52,111 @@ export const ProductCreateView: React.FC = () => {
...
-    </TypedProductCreateMutation>
+    <>
+      <WindowTitle
+        title={intl.formatMessage({
+          defaultMessage: "Create Product",
+          description: "window title"
+        })}
+      />
+      <ProductCreatePage
+        currency={maybe(() => shop.defaultCurrency)}
```
Comment (AlicjaSzu) [inline src/products/views/ProductCreate.tsx:110]: we can remove maybe

File: src/products/views/ProductVariant.tsx [hunk 2]
```diff
@@ -70,132 +79,125 @@ export const ProductVariant: React.FC<ProductUpdateProps> = ({
...
-                      variantId: variant.id
-                    });
-                  }
-                }
-              };
+  const handleImageSelect = (id: string) => () => {
+    if (variant) {
+      if (
+        variant.images &&
+        variant.images.map(image => image.id).indexOf(id) !== -1
```
Comment (AlicjaSzu) [inline src/products/views/ProductVariant.tsx:127]: `variant.images?.map`

**Review by AlicjaSzu — APPROVED** [review]

File: src/products/views/ProductCreate.tsx (hunk 1 above)
Comment (dominik-zeglen) [reply src/products/views/ProductCreate.tsx:110]: More like `shop?.defaultCurrency`,  but yeah

File: src/products/views/ProductVariant.tsx (hunk 2 above)
Comment (dominik-zeglen) [reply src/products/views/ProductVariant.tsx:127]: Didn't touch this, but I'll fix it

**Pull request merged**